micov/_quant.py

```diff
@@ -74,27 +86,26 @@ def pos_to_bins(pos, genome_length, bin_num):
     )
     pos = pos.with_columns([cut_start, cut_stop])
 
-    # Update stop_bin_idx +1 for pl.arange and generate range of bins
+    # update stop_bin_idx +1 for pl.arange and generate range of bins
     pos = pos.with_columns(
         (pl.col("stop_bin_idx") + 1).alias("stop_bin_idx_add1")
     )
 
-    # Generate range of bins covered
+    # generate range of bins covered
     pos = pos.with_columns(
         pl.int_ranges("start_bin_idx", "stop_bin_idx_add1").alias("bin_idx")
     ).drop("stop_bin_idx_add1")
 
-    # Generate bin_df
-    bin_df = (
+    # generate bin_df
+    df_bins = (
         pos.explode("bin_idx")
-        .group_by("bin_idx")
+        .group_by(COLUMN_GENOME_ID, variable, "bin_idx")
         .agg(
             pl.col("start").len().alias("read_hits"),
-            pl.col("sample_id").n_unique().alias("sample_hits"),
-            pl.col("sample_id").unique().sort().alias("samples"),
+            pl.col(COLUMN_SAMPLE_ID).n_unique().alias("sample_hits"),
+            pl.col(COLUMN_SAMPLE_ID).unique().sort().alias("samples"),
         )
-        .sort(by="bin_idx")
+        .sort(by=["bin_idx", variable])
         .join(bin_list, how="left", on="bin_idx")
     )
```

Comment:
Can this be done with a `LazyFrame`? `group_by` and `join` are supported 

Suggested code change isn't allowed on this block, but I think It would be basically:

```
df_bins = (pos.lazy(). ... the other stuff
              .join(bin_list.lazy() ...)
              .collect())

Reply:
yeah everything is lazy here starting from the input